packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_HarmonicCoarseOperator_def.hpp

```diff
@@ -149,149 +140,272 @@ namespace FROSch {
+            interior->buildEntityMap(nodesMap);
+            
+            MultiVectorPtrVecPtr translations = computeTranslations(blockId,interior);
+            for (UN i=0; i<translations.size(); i++) {
+                this->InterfaceCoarseSpaces_[blockId]->addSubspace(interior->getEntityMap(),translations[i]);
+            }
+            if (useRotations) {
+                MultiVectorPtrVecPtr rotations = computeRotations(blockId,dimension,nodeList,interior);
+                for (UN i=0; i<rotations.size(); i++) {
+                    this->InterfaceCoarseSpaces_[blockId]->addSubspace(interior->getEntityMap(),rotations[i]);
+                }
+            }
+            
+            InterfaceCoarseSpaces_[blockId]->assembleCoarseSpace();
+            
+            // Count entities
+            GO numEntitiesGlobal = interior->getEntityMap()->getMaxAllGlobalIndex();
+            if (interior->getEntityMap()->lib()==Xpetra::UseEpetra || interior->getEntityMap()->getGlobalNumElements()>0) {
+                numEntitiesGlobal += 1;
+            }
+            
+            if (this->MpiComm_->getRank() == 0) {
+                std::cout << "\n\
+                --------------------------------------------\n\
+                # volumes:               --- " << numEntitiesGlobal << "\n\
+                --------------------------------------------\n\
+                Coarse space:\n\
+                --------------------------------------------\n\
+                volume: translations      --- " << 1 << "\n\
+                volume: rotations         --- " << useRotations << "\n\
+                --------------------------------------------\n";
+            }
+        }
+        return 0;
+    }
+    
+    template <class SC,class LO,class GO,class NO>
+    typename HarmonicCoarseOperator<SC,LO,GO,NO>::MultiVectorPtrVecPtr HarmonicCoarseOperator<SC,LO,GO,NO>::computeTranslations(UN blockId,
+                                                                                                                                EntitySetPtr entitySet)
+    {
+        MultiVectorPtrVecPtr translations(this->DofsPerNode_[blockId]);
+        MapPtr serialGammaMap = Xpetra::MapFactory<LO,GO,NO>::Build(this->K_->getRangeMap()->lib(),this->GammaDofs_[blockId].size(),0,this->SerialComm_);
+        for (UN i=0; i<this->DofsPerNode_[blockId]; i++) {
+            if (entitySet->getNumEntities()>0) {
+                translations[i] = Xpetra::MultiVectorFactory<SC,LO,GO,NO>::Build(serialGammaMap,entitySet->getNumEntities());
+            } else {
+                translations[i] = Teuchos::null;
+            }
+        }
+        
+        for (UN k=0; k<this->DofsPerNode_[blockId]; k++) {
+            UN ii=0;
+            for (UN i=0; i<entitySet->getNumEntities(); i++) {
+                for (UN j=0; j<entitySet->getEntity(i)->getNumNodes(); j++) {
+                    translations[k]->replaceLocalValue(entitySet->getEntity(i)->getGammaDofID(j,k),ii,1.0);
```

Comment:
We can use `i` of for-loop instead of `ii` in `translations[k]->replaceLocalValue(entitySet->getEntity(i)->getGammaDofID(j,k),ii,1.0);`

Reply:
Thank you for this comment. I completely missed this. I just pushed a small commit to get rid of the ii.